File: java/Ice/plugin/HelloPluginI.java
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

public class HelloPluginI implements com.zeroc.Ice.Plugin
{
    public HelloPluginI(com.zeroc.Ice.Communicator communicator)
    {
        _communicator = communicator;
    }

    @Override
    public void initialize()
    {
        com.zeroc.Ice.ObjectAdapter adapter = _communicator.createObjectAdapter("Hello");
        adapter.add(new HelloI(), com.zeroc.Ice.Util.stringToIdentity("hello"));
        adapter.activate();
    }

    @Override
    public void destroy()
    {
    }

    private com.zeroc.Ice.Communicator _communicator;
}

File: csharp/Ice/plugin/HelloPluginI.cs
// Copyright (c) ZeroC, Inc.

public class HelloPluginI(Ice.Communicator communicator) : Ice.Plugin
{
    public void initialize()
    {
        var adapter = _communicator.createObjectAdapter("Hello");
        adapter.add(new HelloI(), Ice.Util.stringToIdentity("hello"));
        adapter.activate();
    }

    public void destroy()
    {
    }

    private Ice.Communicator _communicator = communicator;
}
